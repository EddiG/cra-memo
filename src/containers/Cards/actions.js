// @flow
import * as constants from './constants';
import type { Card } from './types';

export const updateCards = (cards: Array<Card>) => ({
  type: constants.UPDATE_CARDS,
  payload: {
    cards,
  },
});

export const closeCards = () => ({
  type: constants.CLOSE_CARDS,
  payload: {},
});

export const triggerCard = (cardId: String) => ({
  type: constants.TRIGGER_CARD,
  payload: {
    id: cardId,
  },
});

export const triggerCardAsync = (cardId: String) => ({
  type: constants.TRIGGER_CARD_ASYNC,
  payload: {
    id: cardId,
  },
});

export const validateCards = () => ({
  type: constants.VALIDATE_CARDS,
  payload: {},
});
