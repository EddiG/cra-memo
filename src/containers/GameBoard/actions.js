import * as constants from './constants';

export const generateCards = () => ({
  type: constants.GENERATE_CARDS,
  payload: {},
});

export const refreshCards = () => ({
  type: constants.REFRESH_CARDS,
  payload: {},
});
