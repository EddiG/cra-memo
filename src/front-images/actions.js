import * as constants from './constants';

export const preloadImagesRequire = () => ({
  type: constants.PRELOAD_FRONT_IMAGES_REQUIRE,
  payload: {},
});

export const preloadImagesComplete = () => ({
  type: constants.PRELOAD_FRONT_IMAGES_COMPLETE,
  payload: {},
});
