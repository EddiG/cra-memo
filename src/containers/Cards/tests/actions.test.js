import * as actions from '../actions';

const cards = [
  {
    id: 'card1',
    isOpened: false,
    isGuessed: false,
    frontImageId: 'frontImage1',
    backImageId: 'backImage1',
  },
  {
    id: 'card2',
    isOpened: false,
    isGuessed: false,
    frontImageId: 'frontImage2',
    backImageId: 'backImage2',
  },
];

describe('cards action', () => {
  describe('updateCards', () => {
    it('should return an action UPDATE_CARDS', () => {
      expect(actions.updateCards(cards)).toMatchSnapshot();
    });
  });

  describe('closeCards', () => {
    it('should dispatch an action CLOSE_CARDS', () => {
      expect(actions.closeCards()).toMatchSnapshot();
    });
  });

  describe('triggerCard', () => {
    it('should return an action TRIGGER_CARD', () => {
      expect(actions.triggerCard('card1')).toMatchSnapshot();
    });
  });

  describe('triggerCardAsync', () => {
    it('should return an action TRIGGER_CARD_ASYNC', () => {
      expect(actions.triggerCardAsync('card1')).toMatchSnapshot();
    });
  });
});
