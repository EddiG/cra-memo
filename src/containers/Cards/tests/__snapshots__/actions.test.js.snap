// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cards action closeCards should dispatch an action CLOSE_CARDS 1`] = `
Object {
  "payload": Object {},
  "type": "CLOSE_CARDS",
}
`;

exports[`cards action triggerCard should return an action TRIGGER_CARD 1`] = `
Object {
  "payload": Object {
    "id": "card1",
  },
  "type": "TRIGGER_CARD",
}
`;

exports[`cards action triggerCardAsync should return an action TRIGGER_CARD_ASYNC 1`] = `
Object {
  "payload": Object {
    "id": "card1",
  },
  "type": "TRIGGER_CARD_ASYNC",
}
`;

exports[`cards action updateCards should return an action UPDATE_CARDS 1`] = `
Object {
  "payload": Object {
    "cards": Array [
      Object {
        "backImageId": "backImage1",
        "frontImageId": "frontImage1",
        "id": "card1",
        "isGuessed": false,
        "isOpened": false,
      },
      Object {
        "backImageId": "backImage2",
        "frontImageId": "frontImage2",
        "id": "card2",
        "isGuessed": false,
        "isOpened": false,
      },
    ],
  },
  "type": "UPDATE_CARDS",
}
`;
